localization/pose_initializer/src/pose_initializer/pose_initializer_core.cpp

```diff
@@ -34,12 +35,13 @@ PoseInitializer::PoseInitializer() : Node("pose_initializer")
   output_pose_covariance_ = get_covariance_parameter(this, "output_pose_covariance");
   gnss_particle_covariance_ = get_covariance_parameter(this, "gnss_particle_covariance");
 
-  if (declare_parameter<bool>("gnss_enabled")) {
-    gnss_ = std::make_unique<GnssModule>(this);
-  }
   if (declare_parameter<bool>("ndt_enabled")) {
     ndt_ = std::make_unique<NdtModule>(this);
-    localization_trigger_ = std::make_unique<LocalizationTriggerModule>(this);
+    ndt_localization_trigger_ = std::make_unique<NdtLocalizationTriggerModule>(this);
+  }
+  if (declare_parameter<bool>("gnss_enabled")) {
+    gnss_ = std::make_unique<GnssModule>(this);
+    if (!ndt_) localization_trigger_ = std::make_unique<LocalizationTriggerModule>(this);
```

Comment:
Why does `ndt_localization_trigger` contain the service for EKF? I think it would be easier to separate the modules.

Reply:
You are right. It has been separated.